client/actions/show-heartbeat/index.js

```diff
@@ -21,11 +30,25 @@ export default class ShowHeartbeatAction extends Action {
     } = this.recipe.arguments;
 
     const lastShown = await this.getLastShownDate();
+
+    // get the last shown for heartbeats in general
+    const lastHeartbeatShown = await this.getLastHeartbeatDate();
+
+    // we can show the heartbeat if there has not been one shown recently,
+    const canShowHeartbeat = lastHeartbeatShown === null
+    // or if the last one shown was more than HEARTBEAT_THROTTLE ms ago
+      || Date.now() - lastHeartbeatShown >= HEARTBEAT_THROTTLE;
+
+    // the survey should display itself if..
     const shouldShowSurvey = (
-      this.normandy.testing
-      || lastShown === null
-      || Date.now() - lastShown > LAST_SHOWN_DELAY
+      // ..we're testing, or..
+      this.normandy.testing ||
```

Comment:
I think the `||` should start out lines rather than end them, that's a little clearer.